[HTTP Robot Lang] Query string with multiple parameters throws Syntax error

# Client Script
connect http://localhost:8080/path?.kr=1&.kb=256
connected

# Server Script
accept http://localhost:8080/path?.kr=1&.kb=256
accepted
connected

# Exception Thrown

line 24:72 token recognition error at: '&'
line 24:75 token recognition error at: '='
line 24:73 mismatched input 'kb' expecting {'bound', 'close', 'closed', 'connected', 'disconnected', 'opened', 'read', 'unbound', 'write'}
line 64:72 token recognition error at: '&'
line 64:75 token recognition error at: '='
line 64:73 mismatched input 'kb' expecting {'bound', 'close', 'closed', 'connected', 'disconnected', 'opened', 'read', 'unbound', 'write'}
line 99:71 token recognition error at: '&'
line 99:74 token recognition error at: '='
line 99:72 mismatched input 'kb' expecting {<EOF>, 'accept', 'accepted', 'bound', 'child', 'close', 'closed', 'connect', 'opened', 'property', 'read', 'unbind', 'unbound', 'write'}
line 141:71 token recognition error at: '&'
line 141:74 token recognition error at: '='
line 141:72 mismatched input 'kb' expecting {<EOF>, 'accept', 'accepted', 'bound', 'child', 'close', 'closed', 'connect', 'opened', 'property', 'read', 'unbind', 'unbound', 'write'}

[ERROR] Caught exception trying to parse script. Sending error to client
org.kaazing.k3po.lang.parser.ScriptParseException: org.kaazing.k3po.lang.parser.ScriptParseException: Syntax error while parsing: 
	at org.kaazing.k3po.lang.parser.ScriptParserImpl.parse(ScriptParserImpl.java:75)

--- a/lang/src/main/antlr4/org/kaazing/k3po/lang/parser/v2/Robot.g4
+++ b/lang/src/main/antlr4/org/kaazing/k3po/lang/parser/v2/Robot.g4
@@ -477,7 +477,7 @@ HttpVersionKeyword
 // URI cannot begin with any of our data type delimiters, and MUST contain a colon.
 URILiteral
     : Letter (Letter | '+')+ ':'
-      (Letter | ':' | ';' | '/' | '=' | '.' | DecimalLiteral | '?' | '%' | '-' | ',' | '*')+
+      (Letter | ':' | ';' | '/' | '=' | '.' | DecimalLiteral | '?' | '&' | '%' | '-' | ',' | '*')+
 //      ~('"' | '/' | ']' | '}')
     ;
 

--- a/lang/src/test/java/org/kaazing/k3po/lang/parser/ScriptParserImplTest.java
+++ b/lang/src/test/java/org/kaazing/k3po/lang/parser/ScriptParserImplTest.java
@@ -1137,14 +1137,20 @@ public class ScriptParserImplTest {
     public void shouldParseConnectScript() throws Exception {
 
         String script =
-                "# tcp.client.connect-then-close\n" + "connect tcp://localhost:7788\n" + "connected\n" + "close\n" + "closed\n";
+                "# tcp.client.connect-then-close\n" +
+                "connect http://localhost:8080/path?p1=v1&p2=v2\n" + 
+                "connected\n" +
+                "close\n" +
+                "closed\n";
 
         ScriptParserImpl parser = new ScriptParserImpl();
         AstScriptNode actual = parser.parseWithStrategy(script, SCRIPT);
 
-        AstScriptNode expected =
-                new AstScriptNodeBuilder().addConnectStream().setLocation(URI.create("tcp://localhost:7788")).addConnectedEvent()
-                        .done().addCloseCommand().done().addClosedEvent().done().done().done();
+        AstScriptNode expected = new AstScriptNodeBuilder()
+                .addConnectStream()
+                .setLocation(URI.create("http://localhost:8080/path?p1=v1&p2=v2"))
+                .addConnectedEvent().done().addCloseCommand().done()
+                .addClosedEvent().done().done().done();
 
         assertEquals(expected, actual);
     }

--- a/lang/src/test/java/org/kaazing/k3po/lang/parser/ScriptParserImplTest.java
+++ b/lang/src/test/java/org/kaazing/k3po/lang/parser/ScriptParserImplTest.java
@@ -1138,7 +1138,7 @@ public class ScriptParserImplTest {
 
         String script =
                 "# tcp.client.connect-then-close\n" +
-                "connect http://localhost:8080/path?p1=v1&p2=v2\n" + 
+                "connect http://localhost:8080/path?p1=v1&p2=v2\n" +
                 "connected\n" +
                 "close\n" +
                 "closed\n";

